### PR TITLE
Added new motion state flags to show an additional fields

### PR DESCRIPTION
**Pull request by emanuelschuetze**
to extend state name and recommendation label

### CODE REVIEW

File: openslides/motions/static/js/motions/base.js [hunk 1]
```diff
@@ -288,6 +289,37 @@ angular.module('OpenSlidesApp.motions', [
...
+                // full state name - optional with custom state name extension
+                // depended by state and provided by a custom comment field
+                getStateName: function () {
+                    var additionalName = '';
+                    if (this.state.show_state_extension_field) {
+                        // check motion comment fields for flag 'forState'
+                        var fields = Config.get('motions_comments').value;
+                        for (var i = 0; i < fields.length; i++) {
+                            if (fields[i].forState) {
+                                additionalName = ' ' + this.comments[i];
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/base.js]: You should break the loop if the first entry was found. Better use [_.findIndex()](https://lodash.com/docs/4.16.4#findIndex).

File: openslides/motions/static/js/motions/base.js [hunk 2]
```diff
@@ -288,6 +289,37 @@ angular.module('OpenSlidesApp.motions', [
...
+                    return gettextCatalog.getString(this.state.name) + additionalName;
+                },
+                // full recommendation string - optional with custom recommendationextension
+                // depended by state and provided by a custom comment field
+                getRecommendationName: function () {
+                    var additionalName = '';
+                    if (this.recommendation.show_recommendation_extension_field) {
+                        // check motion comment fields for flag 'forRecommendation'
+                        var fields = Config.get('motions_comments').value;
+                        for (var i = 0; i < fields.length; i++) {
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/base.js]: Here the same.

File: openslides/motions/static/js/motions/base.js [hunk 3]
```diff
@@ -480,7 +512,17 @@ angular.module('OpenSlidesApp.motions', [
                 for (var i = 0; i < fields.length; i++) {
                     motion.comments.push(motion['comment ' + fields[i].name] || '');
                 }
-            }
+            },
+            getFieldNameForFlag: function (flag) {
+                var fields = Config.get('motions_comments').value;
+                var fieldName = '';
+                for (var i = 0; i < fields.length; i++) {
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/base.js]: Here the same.

File: openslides/motions/static/js/motions/site.js [hunk 4]
```diff
@@ -1026,7 +1027,7 @@ angular.module('OpenSlidesApp.motions.site', [
     'ProjectionDefault',
     function($scope, $http, operator, ngDialog, MotionForm,
              ChangeRecommmendationCreate, ChangeRecommmendationView, MotionChangeRecommendation, MotionPDFExport,
-             Motion, Category, Mediafile, Tag, User, Workflow, Config, motion, MotionInlineEditing, gettextCatalog,
+             Motion, MotionComment, Category, Mediafile, Tag, User, Workflow, Config, motion, MotionInlineEditing, gettextCatalog,
              Projector, ProjectionDefault) {
         Motion.bindOne(motion.id, $scope, 'motion');
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/site.js]: Remove this line.

File: openslides/motions/static/js/motions/site.js [hunk 5]
```diff
@@ -1041,10 +1042,17 @@ angular.module('OpenSlidesApp.motions.site', [
         }, function () {
             $scope.defaultProjectorId = ProjectionDefault.filter({name: 'motions'})[0].projector_id;
         });
+        $scope.$watch(function () {
+            return Motion.lastModified();
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/site.js]: Use motion.id in the lastModified check.

File: openslides/motions/static/js/motions/site.js [hunk 6]
```diff
@@ -1119,6 +1127,26 @@ angular.module('OpenSlidesApp.motions.site', [
...
             $http.put('/rest/motions/motion/' + motion.id + '/set_state/', {});
         };
+        // save additional state field
+        $scope.saveAdditionalStateField = function (stateExtension) {
+            if (stateExtension) {
+                motion["comment " + $scope.commentFieldForState] = stateExtension;
+                motion.title = motion.getTitle(-1);
+                motion.text = motion.getText(-1);
+                motion.reason = motion.getReason(-1);
+                Motion.save(motion);
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/site.js]: Under circumstances this creates a new motion version. Do you want to do so only to set the comment field name for the state or recommendation?

File: openslides/motions/static/js/motions/site.js [hunk 7]
```diff
@@ -1119,6 +1127,26 @@ angular.module('OpenSlidesApp.motions.site', [
...
+            }
+        };
+        // save additional recommendation field
+        $scope.saveAdditionalRecommendationField = function (recommendationExtension) {
+            if (recommendationExtension) {
+                motion["comment " + $scope.commentFieldForRecommendation] = recommendationExtension;
+                motion.title = motion.getTitle(-1);
+                motion.text = motion.getText(-1);
+                motion.reason = motion.getReason(-1);
+                Motion.save(motion);
```
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/site.js]: Here the same.

File: openslides/motions/static/js/motions/site.js (hunk 6 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/site.js]: yes, this function should save only the additional state (or recommendationI) comment field. But the server requires that title, text and reason is set to save other values (like comments). Do you have any hint to improve it?

File: openslides/motions/static/js/motions/site.js (hunk 7 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/site.js]: see above

File: openslides/motions/static/js/motions/base.js (hunk 1 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/base.js]: done

File: openslides/motions/static/js/motions/base.js (hunk 2 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/base.js]: done

File: openslides/motions/static/js/motions/base.js (hunk 3 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/base.js]: done

File: openslides/motions/static/js/motions/site.js (hunk 4 above)
Comment (emanuelschuetze) [inline openslides/motions/static/js/motions/site.js]: done

File: openslides/motions/static/js/motions/site.js (hunk 6 above)
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/site.js]: Yes: either we refactor the REST API at this point or you add a new viewpoint like set_state.

**Comment by normanjaeckel** [comment]
Ok. The save() problem is still open.

**Comment by normanjaeckel** [comment]
Please open issue (and assign it to me) with the save() problem. Then merge this.

**Comment by emanuelschuetze** [comment]
Issue created for refactor save method: #2571

**Pull request merged**